validation/src/test/scala/hmda/validation/rules/ts/validity/V140Spec.scala

```diff
@@ -0,0 +1,31 @@
+package hmda.validation.rules.ts.validity
+
+import hmda.parser.fi.ts.TsGenerators
+import hmda.validation.dsl.{ Failure, Success }
+import org.scalatest.{ MustMatchers, PropSpec }
+import org.scalatest.prop.PropertyChecks
+
+class V140Spec extends PropSpec with PropertyChecks with MustMatchers with TsGenerators with ValidityUtils {
+
+  property("Respondent state code must equal a valid postal code abbreviation") {
+    forAll(tsGen) { ts =>
+      val r = ts.respondent
+      whenever(respondentNotEmpty(r)) {
+        V140(ts) mustBe Success()
+      }
+    }
+  }
+
+  property("Wrong or missing respondent state code should fail") {
+    forAll(tsGen) { ts =>
+      val r1 = ts.respondent.copy(state = "")
+      val badTs1 = ts.copy(respondent = r1)
+      val r2 = ts.respondent.copy(state = "XXX")
```

Comment:
It'd be nice to have a test (here or elsewhere) covering the case of a 2-digit string that isn't a valid state code. (I'm not gonna insist on it before merging, but it's nice to have a minimally-different example.)